source/client/benchmark_client_impl.h

```diff
@@ -66,6 +67,73 @@ class Http1PoolImpl : public Envoy::Http::Http1::ProdConnPoolImpl {
   bool prefetch_connections_{};
 };
 
+// Vanilla Envoy's HTTP/2 pool is single connection only (or actually sometimes dual in connection
+// drainage scenarios). Http2PoolImpl is an experimental pool, which uses multiple vanilla Envoy
+// HTTP/2 pools under the hood. Using multiple connections is useful when testing backends that need
+// multiple connections to distribute load internally. Combining multiple connections with
+// --max-requests-per-connection may help as well, as doing periodically initiating new connections
+// may help the benchmark target by giving it an opportunity to rebalance.
+class Http2PoolImpl : public Envoy::Http::ConnectionPool::Instance,
+                      public Envoy::Http::ConnPoolImplBase {
+public:
+  Http2PoolImpl(
+      Envoy::Event::Dispatcher& dispatcher, Envoy::Upstream::HostConstSharedPtr host,
+      Envoy::Upstream::ResourcePriority priority,
+      const Envoy::Network::ConnectionSocket::OptionsSharedPtr& options,               // NOLINT
+      const Envoy::Network::TransportSocketOptionsSharedPtr& transport_socket_options) // NOLINT
+      : Envoy::Http::ConnPoolImplBase(std::move(host), priority), dispatcher_(dispatcher),
+        socket_options_(options), transport_socket_options_(transport_socket_options) {
+    for (uint32_t i = 0; i < host_->cluster().resourceManager(priority_).connections().max(); i++) {
```

Comment:
is there a preference here for having this implementation in the header file, rather than the cc file?

Reply:
My personal thinking is that even though this and the other functions in this file are all fairly small, it makes it harder to get a clean look at the api in general, but I'm open to other opinions.

Reply:
No good reason, so module one liners (getters/setters) which I feel don't really interfere with readablility of the api, I moved implementations to the `.cc`. Wdyt?

Reply:
that makes sense to me. Happy to follow your instincts on this in general. I do think the api is clearer for me now though, so thank you for doing that!